uniffi_bindgen/src/bindings/gecko/templates/InterfaceHeaderTemplate.h

```diff
@@ -0,0 +1,55 @@
+// This file was autogenerated by some hot garbage in the `uniffi` crate.
+// Trust me, you don't want to mess with it!
+
+{% import "macros.cpp" as cpp %}
+
+#ifndef mozilla_dom_{{ obj.name()|class_name_webidl }}
+#define mozilla_dom_{{ obj.name()|class_name_webidl }}
+
+#include "jsapi.h"
+#include "nsCOMPtr.h"
+#include "nsIGlobalObject.h"
+#include "nsWrapperCache.h"
+
+#include "mozilla/RefPtr.h"
+
+#include "mozilla/dom/{{ ci.namespace()|class_name_webidl }}Shared.h"
+
+namespace mozilla {
+namespace dom {
+
+class {{ obj.name()|class_name_cpp }} final : public nsISupports, public nsWrapperCache {
+ public:
+  NS_DECL_CYCLE_COLLECTING_ISUPPORTS
+  NS_DECL_CYCLE_COLLECTION_SCRIPT_HOLDER_CLASS({{ obj.name()|class_name_cpp }})
```

Comment:
Technically, since we only ever return new objects, we might not need cycle collection...but we do need to cycle-collect `mGlobal` if we hold onto it, and we may need it once we start returning references, _and_ also all this might go away and get replaced with XPIDL—so for now, copypasta it is! 🍝😁